packages/components/src/unit-control/styles/unit-control-styles.ts

```diff
@@ -18,22 +18,13 @@ type SelectProps = {
 
 type InputProps = {
 	disableUnits?: boolean;
-	size: SelectSize;
 };
 
 export const Root = styled.div`
 	box-sizing: border-box;
 	position: relative;
 `;
 
-const paddingStyles = ( { disableUnits }: InputProps ) => {
-	if ( disableUnits ) return '';
-
-	return css`
-		${ rtl( { paddingRight: 8 } )() };
-	`;
-};
-
```

Comment:
My memory is failing me on this one, I don't recall any blockers. It might have been as simple as I didn't see the `prefix` prop in the `UnitControl` readme and didn't follow the props through to the underlying `InputControl`. 

It also looks like we might need to tweak the `UnitControl` types, as attempting to pass the border control dropdown via a `prefix` prop on the `UnitControl` results in a type error that it expects `prefix` to be `string | undefined`.

I'd be happy to work on rendering the border control's dropdown as a prefix in a follow up to this PR.

Reply:
That sounds great, thanks! It's nice to see things getting less hacky as we converge on common patterns 🙂

Reply:
I've put together a PR (https://github.com/WordPress/gutenberg/pull/42212) that now renders the dropdown via the `UnitControl` prefix and attempts to clean up the styles. There could still be more to polish on that control when it comes time to add a large 40px size variant.